pyproject.toml

```diff
@@ -40,52 +40,53 @@ classifiers = [
 ]
 dynamic = ["description", "version"]
 dependencies = [
-  "bottleneck >=1.3.1",
-  # This is differently named in conda-forge
-  "cf-xarray >=0.6.1",
-  "cftime >=1.4.1",
+  "bottleneck >=1.3.1", # Is it needed? I think that dask already uses it as a dependence, and it's not used in the code
+  "cf-xarray >=0.6.1", # Is it needed? It's not used in the code, nor is it in environment-dev.yml
+  "cftime >=1.4.1", # Same as above
   "clisops >=0.9.2",
   "dask[array] >=2.6",
   "dask-geopandas >=0.4.1",
   "geopandas >=1.0",
-  "intake <2.0.0", # Pinned until catalogs support v2.0 conventions. See: https://github.com/hydrologie/xdatasets/issues/82",
-  "intake-xarray >=0.6.1,<2.0.0",
+  "intake",
   "intake-geopandas",
-  "ipython >=8.5.0",
+  "intake-xarray >=0.6.1,<2.0.0", # FIXME: This needs to be updated to >=2.0.0 to work with Python 3.13. One of the dependencies of <2.0.0 (distutils) is already deprecated in Python 3.12, but accessible through setuptools/wheel.
+  "ipython >=8.5.0", # This should probably not be in the pyproject.toml file
   "jsonpickle",
   "numpy >=1.23",
   "pandas >=1.5",
+  "setuptools", # Needed for Python 3.12 until intake-xarray is updated
   "s3fs",
   "scipy >=1.9.0",
   "tqdm",
   "xarray >=2023.1",
+  "wheel", # Needed for Python 3.12 until intake-xarray is updated
   "zarr >=2.11.1"
 ]
 
 [project.optional-dependencies]
 dev = [
   # Dev tools and testing
+  "pip >=25.0",
+  "bump-my-version >=0.30.1",
   "black ==25.1.0",
   "blackdoc ==0.3.9",
-  "bump-my-version >=0.30.1",
   "coverage >=7.5.0",
   "flake8 >=7.1.1",
   "flake8-rst-docstrings >=0.3.0",
   "flit >=3.10.1,<4.0",
   "isort ==6.0.1",
   "mypy >=1.14.1",
-  "nbval",
+  "nbval", # Is it needed? It's not in environment-dev.yml
   "numpydoc >=1.8.0",
-  "pip >=25.0",
   "pre-commit >=3.5.0",
-  "pylint",
-  "pytest-cov >=5.0.0",
+  "pylint", # Is it needed? It's not in environment-dev.yml
```

Comment:
This might be needed for CI linting. Could be added to `environment-dev.yml` as well.